fuel/datasets/hdf5.py

```diff
@@ -106,16 +106,20 @@ class H5PYDataset(Dataset):
          array
       4. ``stop`` : stop index (exclusive) of the split in the source
          array
-      5. ``available`` : boolean, ``False`` is this split is not available
+      6. ``indices`` : h5py.Reference, reference to a dataset containing
+         subset indices for this split/source pair.
+      6. ``available`` : boolean, ``False`` is this split is not available
          for this source
-      6. ``comment`` : comment string
+      7. ``comment`` : comment string
 
     Parameters
     ----------
     file_or_path : :class:`h5py.File` or str
         HDF5 file handle, or path to the HDF5 file.
-    which_set : str
-        Which split to use.
+    which_set : str or tuple of str
```

Comment:
I'm not very opinionated on that, but if it helps uniformize the interface in Blocks and Fuel I'd be for it. Since we're already in the process of changing the interface, I guess there's no harm in taking that decision now.

However, since a string is itself an iterable and the current interface expects a string, I'm wary of introducing silent bugs where the user passes `'train'` and the code understands that as five splits, `'t'`, `'r'`, `'a'`, `'i'` and `'n'`. Except for type checking and raising an informative error message if the user passes a string, I don't see a good way of making sure this doesn't happen. What are your thoughts on this?